expression/builtin_time.go

```diff
@@ -4561,13 +4572,33 @@ func (b *builtinMakeTimeSig) Clone() builtinFunc {
 func (b *builtinMakeTimeSig) evalDuration(row chunk.Row) (types.Duration, bool, error) {
 	dur := types.ZeroDuration
 	dur.Fsp = types.MaxFsp
-	hour, isNull, err := b.args[0].EvalInt(b.ctx, row)
-	if isNull || err != nil {
-		return dur, isNull, errors.Trace(handleInvalidTimeError(b.ctx, err))
+	var hour int64
+	if b.args[0].GetType().EvalType() == types.ETReal {
```

Comment:
Seems that we can extract following part as a function, then call it for `b.args[0]` and `b.args[1]`?